src/unity/python/turicreate/visualization/plot.py

```diff
@@ -0,0 +1,29 @@
+import time
+import json
+
+class Plot(object):
+    def __init__(self):
+        self.plot_object = tc.extensions._Plot()
+
+    def show(self):
+        self.plot_object.show()
+
+    def save_vega(self, filepath, include_data=True):
+        file_contents = self.plot_object.get_vega(include_data)
+
+        with open(filepath, 'w') as fp:
+            json.dump(file_contents, fp)
```

Comment:
Doesn't `get_vega` return an `std::string`? In that case you should just use `fp.write(file_contents)`.

Reply:
get_vega returns a json string since, it's easier to include the data on the Python side, rather than shoving the data into the vega_spec on the C++ side. So I parse the vega, parse the data and stitch the two together on the front end.